vcd/datasource_vcd_storage_profile.go

```diff
@@ -0,0 +1,55 @@
+package vcd
+
+import (
+	"context"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func datasourceVcdStorageProfile() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdStorageProfileRead,
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
```

Comment:
I hit something very strange while testing this.

Reproduction:
1. Added a 2nd storage profile with the name "Development" to my VDC.
2. Added the following to an existing HCL:
```
data "vcd_storage_profile" "development" {
  name = "Development"
}
output "storage_profile_id" {
  value = data.vcd_storage_profile.development.id
}
```

To my big surprise, when I ran `terraform apply` it suggested to re-create (!) the VDC. Could you try to reproduce this?

```
$ terraform apply
data.vcd_resource_list.org_list: Refreshing state... [id=list_of_orgs]
data.vcd_storage_profile.development: Refreshing state...
data.vcd_resource_list.vdc_list: Refreshing state... [id=list_of_orgs]
data.vcd_resource_schema.org_struct: Refreshing state... [id=org_struct]
vcd_org.shared_org: Refreshing state... [id=urn:vcloud:org:81b36c83-74a6-4e69-a9c2-100310d46f65]
vcd_org_vdc.shared_vdc: Refreshing state... [id=urn:vcloud:vdc:06905309-97bf-4876-9503-2b90f17fc5da]
vcd_catalog.shared_catalog: Refreshing state... [id=urn:vcloud:catalog:834cc462-e36f-47de-91db-95740f58026a]
vcd_catalog_item.bitnami_wordpress: Refreshing state... [id=urn:vcloud:catalogitem:465d29f3-1e2d-41eb-82a1-cfe3f4cb78db]
vcd_catalog_item.bitnami_nginx: Refreshing state... [id=urn:vcloud:catalogitem:279c9ac0-e96f-4c9c-b4e0-801bbd6c8fcc]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # vcd_org_vdc.shared_vdc must be replaced
-/+ resource "vcd_org_vdc" "shared_vdc" {
        allocation_model            = "Flex"
      + allow_over_commit           = (known after apply)
        cpu_guaranteed              = 0.92
        cpu_speed                   = 1000
      ~ default_vm_sizing_policy_id = "urn:vcloud:vdcComputePolicy:81448202-dee7-4fd4-8a28-5e91c1b61fdb" -> (known after apply)
        delete_force                = true
        delete_recursive            = true
        description                 = "Terraform created VDC"
        elasticity                  = true
        enable_fast_provisioning    = true
        enable_thin_provisioning    = true
      - enable_vm_discovery         = false -> null
        enabled                     = true
      ~ id                          = "urn:vcloud:vdc:06905309-97bf-4876-9503-2b90f17fc5da" -> (known after apply)
        include_vm_memory_overhead  = true
        memory_guaranteed           = 0.91
      - metadata                    = {} -> null
        name                        = "shared-vdc"
        network_pool_name           = "vc1-TestbedCluster-13:07:00-VXLAN-NP"
        network_quota               = 5
      - nic_quota                   = 0 -> null
        provider_vdc_name           = "vc1-TestbedCluster-13:07:00"
      - vm_quota                    = 0 -> null
      ~ vm_sizing_policy_ids        = [
          - "urn:vcloud:vdcComputePolicy:81448202-dee7-4fd4-8a28-5e91c1b61fdb",
        ] -> (known after apply)

      ~ compute_capacity {
          ~ cpu {
                allocated = 3000
                limit     = 8000
              ~ reserved  = 2760 -> (known after apply)
              ~ used      = 0 -> (known after apply)
            }

          ~ memory {
                allocated = 4000
              ~ limit     = 4000 -> (known after apply)
              ~ reserved  = 3640 -> (known after apply)
              ~ used      = 0 -> (known after apply)
            }
        }

      - storage_profile { # forces replacement
          - default            = false -> null
          - enabled            = true -> null
          - limit              = 0 -> null
          - name               = "Development" -> null
          - storage_used_in_mb = 0 -> null
        }
      - storage_profile { # forces replacement
          - default            = true -> null
          - enabled            = true -> null
          - limit              = 0 -> null
          - name               = "*" -> null
          - storage_used_in_mb = 20480 -> null
        }
      + storage_profile { # forces replacement
          + default            = true
          + enabled            = true
          + limit              = 0
          + name               = "*"
          + storage_used_in_mb = (known after apply)
        }
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Changes to Outputs:
  + storage_profile_id = "urn:vcloud:vdcstorageProfile:020433e5-9186-41ff-9a34-df146b25c52c"

Do you want to perform these actions?
```

Reply:
Please ignore the above. As I added the 2nd storage profile in UI, I missed to add it into HCL.